gabriel-portfolio/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,88 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-// Did not remove this because it will be used in week 3
-// @WebServlet("/data")
-// public class DataServlet extends HttpServlet {
-//   @Override
-//   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-//     response.setContentType("text/html;");
-//     response.getWriter().println("<h1>Hello world!</h1>");
-//   }
-// }
+
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+  
+  final String commentEntity = "Comment";
+  final String textProperty = "text";
+  final String timestampProperty = "timestamp";
+
+  static Gson gson;
+
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query(commentEntity).addSort(timestampProperty, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // loop through the query to get the text from each comment
```

Comment:
nit: Please format comments like sentences (capitalize and punctuation) here and below.